File: funnyvote/app/src/main/java/com/heaton/funnyvote/ui/about/AboutContract.java
package com.heaton.funnyvote.ui.about;

import com.heaton.funnyvote.BasePresenter;
import com.heaton.funnyvote.BaseView;

public interface AboutContract {
    interface Presenter extends BasePresenter{
        void IntentToIntroduction();
        void IntentToAuthorInfo();
        void IntentToLicence();
        void IntentToProblem();
        void IntentToAppStore();
        void IntentToAbout();
        void IntentToShareApp();
    }
    interface View extends BaseView<Presenter> {
        void showIntroduction();
        void showAuthorInfo();
        void showLicence();
        void showProblem();
        void showAppStore();
        void showAbout();
        void showShareApp();
    }
}

File: funnyvote/app/src/main/java/com/heaton/funnyvote/ui/about/AboutContract.kt
package com.heaton.funnyvote.ui.about

import com.heaton.funnyvote.BasePresenter
import com.heaton.funnyvote.BaseView

interface AboutContract {
    interface Presenter : BasePresenter {
        fun IntentToIntroduction()
        fun IntentToAuthorInfo()
        fun IntentToLicence()
        fun IntentToProblem()
        fun IntentToAppStore()
        fun IntentToAbout()
        fun IntentToShareApp()
    }

    interface View : BaseView<Presenter> {
        fun showIntroduction()
        fun showAuthorInfo()
        fun showLicence()
        fun showProblem()
        fun showAppStore()
        fun showAbout()
        fun showShareApp()
    }
}
